pandas/core/internals/array_manager.py

```diff
@@ -452,7 +452,11 @@ def putmask(self, mask, new, align: bool = True):
         )
 
     def diff(self, n: int, axis: int) -> ArrayManager:
-        return self.apply_with_block("diff", n=n, axis=axis)
+        if axis == 1:
+            axis = 0
+        elif axis == 0:
+            assert n == 0
```

Comment:
comment on why this is expected (presumably the caller checks?)

Reply:
Yeah, DataFrame diff already special cases that (and basically "misuses" Manager.diff with a wrong axis for `n=0`, in which case the axis doesn't matter). 
Will add a comment for that.